torch/nn/parallel/distributed.py

```diff
@@ -539,7 +539,15 @@ def __init__(
         super(DistributedDataParallel, self).__init__()
         Joinable.__init__(self)
         self.logger = None
-        if not any((p.requires_grad for p in module.parameters())):
+
+        if hasattr(module, "_ddp_params_and_buffers_to_ignore"):
+            self.parameters_to_ignore = module._ddp_params_and_buffers_to_ignore
```

Comment:
Do we need to make this into a (public) attribute? Can we also make it into a `set` given the membership checks for constructing `module_parameters`?
```suggestion
            parameters_to_ignore = set(module._ddp_params_and_buffers_to_ignore)
```